OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc

```diff
@@ -1153,19 +1153,190 @@ fmi2Status fmi2SetString(fmi2Component c, const fmi2ValueReference vr[], size_t
   return fmi2OK;
 }
 
+/*
+* print information about ring buffer simulation data
+*/
+void printRingBufferSimulationData(RINGBUFFER *rb, DATA* data)
+{
+
+  for (int i = 0; i < ringBufferLength(rb); i++)
+  {
+    messageClose(LOG_STDOUT);
+    SIMULATION_DATA *sdata = (SIMULATION_DATA *)getRingData(rb, i);
+    infoStreamPrint(LOG_STDOUT, 1, "Time: %g ", sdata->timeValue);
+
+    infoStreamPrint(LOG_STDOUT, 1, "RingBuffer Real Variable");
+    for (int j = 0; j < data->modelData->nVariablesReal; ++j)
+    {
+      infoStreamPrint(LOG_STDOUT, 0, "%ld: %s = %g ", j+1, data->modelData->realVarsData[j].info.name, sdata->realVars[j]);
+    }
+    messageClose(LOG_STDOUT);
+
+    infoStreamPrint(LOG_STDOUT, 1, "RingBuffer Integer Variable");
+    for (int j = 0; j < data->modelData->nVariablesInteger; ++j)
+    {
+      infoStreamPrint(LOG_STDOUT, 0, "%ld: %s = %i ", j+1, data->modelData->integerVarsData[j].info.name, sdata->integerVars[j]);
+    }
+    messageClose(LOG_STDOUT);
+
+    infoStreamPrint(LOG_STDOUT, 1, "RingBuffer Boolean Variable");
+    for(int j = 0; j < data->modelData->nVariablesBoolean; ++j)
+    {
+      infoStreamPrint(LOG_STDOUT, 0, "%ld: %s = %s ", j+1, data->modelData->booleanVarsData[j].info.name, sdata->booleanVars[j] ? "true" : "false");
+    }
+    messageClose(LOG_STDOUT);
+  }
+}
+
 fmi2Status fmi2GetFMUstate(fmi2Component c, fmi2FMUstate* FMUstate)
 {
-  return unsupportedFunction(c, "fmi2GetFMUstate", modelInstantiated|modelInitializationMode|modelEventMode|modelContinuousTimeMode|modelTerminated|modelError);
+  ModelInstance *comp = (ModelInstance *) c;
+  fmi2CallbackFunctions* functions = (fmi2CallbackFunctions*) comp->functions;
+  int meStates = modelInstantiated|modelInitializationMode|modelEventMode;
+  int csStates = modelInstantiated|modelInitializationMode|modelEventMode|modelContinuousTimeMode;
+  if (invalidState(comp, "fmi2GetFMUstate", meStates, csStates))
+    return fmi2Error;
+
+  INTERNAL_FMU_STATE* internal_state = (INTERNAL_FMU_STATE*) functions->allocateMemory(1, sizeof(INTERNAL_FMU_STATE));
+  internal_state->simulationData = allocRingBuffer(SIZERINGBUFFER, sizeof(SIMULATION_DATA));
+
+  DATA* fmudata = (DATA *) comp->fmuData;
+
+  /* prepare RingBuffer, by allocating memory for real Vars
+   * copy the ring buffer data to INTERNAL_FMU_STATE
+  */
+  SIMULATION_DATA tmpSimData = {0};
+  for (int i = 0; i < ringBufferLength(fmudata->simulationData); i++)
+  {
+    tmpSimData.timeValue = fmudata->localData[i]->timeValue;
+    /* allocate memory for all Real variables */
+    tmpSimData.realVars = (modelica_real *)calloc(fmudata->modelData->nVariablesReal, sizeof(modelica_real));
```

Comment:
Us `functions->allocateMemory` instead of `calloc`.

Reply:
this is fixed with this commit https://github.com/OpenModelica/OpenModelica/pull/8638/commits/099afca47fd4233e485356c16cbf416a03a2cf1c